apis/externalsecrets/v1beta1/externalsecret_validator.go

```diff
@@ -61,6 +61,14 @@ func validateExternalSecret(obj runtime.Object) (admission.Warnings, error) {
 		if findOrExtract && ref.SourceRef != nil && ref.SourceRef.GeneratorRef != nil {
 			errs = errors.Join(errs, fmt.Errorf("generator can not be used with find or extract"))
 		}
+
+		if ref.SourceRef != nil &&  ref.SourceRef.GeneratorRef == nil {
+			errs = errors.Join(errs, fmt.Errorf("generator must be set if SourceRef is used"))
+		}
+
+		if !findOrExtract && (ref.SourceRef != nil ||  ref.SourceRef.GeneratorRef != nil) {
```

Comment:
This condition seems to be wrong. It should be  

```
!findOrExtract && (ref.SourceRef == nil || ref.SourceRef.GeneratorRef == nil)
```

Also, I found that `!findOrExtract` is hard to understand. Why don't we just use the following condition?

```
ref.Find == nil && ref.Extract == nil && (ref.SourceRef == nil || ref.SourceRef.GeneratorRef == nil)
```

Reply:
Would you add test cases to cover the four cases?